### PR TITLE
Updates to make plugin work with next Redmine V4.0.0

### PR DESCRIPTION
**Pull request by Utopism**
Hi,

Can you please review this pull request ?
This version is compatible with previous versions.

Thanks a lot.

### CODE REVIEW

**Comment by kzalewski** [comment]
I can confirm that this patch works nicely with Redmine 4.0.0.  Without it, a migration to Redmine v4.0 produces a bunch of errors when executing `rake db:migrate`.

**Comment by Utopism** [comment]
Thank kzalewski to have tested it.

peclick, can you merge this pull request ?

**Comment by canyildiz** [comment]
This worked great with the following environment:

```
Environment:
  Redmine version                4.0.1.stable
  Ruby version                   2.5.3-p105 (2018-10-18) [x86_64-linux]
  Rails version                  5.2.2
  Environment                    production
  Database adapter               PostgreSQL
  Mailer queue                   ActiveJob::QueueAdapters::AsyncAdapter
  Mailer delivery                smtp
SCM:
  Subversion                     1.9.5
  Mercurial                      4.0
  Bazaar                         2.8.0
  Git                            2.11.0
  Filesystem                     
Redmine plugins:
  clipboard_image_paste          1.13
```

File: lib/clipboard_image_paste/attachment_patch.rb [hunk 1]
```diff
@@ -13,34 +13,17 @@
 #*******************************************************************************
 
 module AttachmentPatch
-  def self.included(base)
-    base.send(:include, InstanceMethods)
-
-    # Same as typing in the class
-    base.class_eval do
-      # Send unloadable so it will not be unloaded in development
-      unloadable
```
Comment (kou) [inline lib/clipboard_image_paste/attachment_patch.rb]: We can remove this because we call `require_dependency` in `to_prepare` block.

**Comment by alfal** [comment]
Great ,this plugin can run again, thanks  @Utopism Utopism && @kou kou

**Comment by bviktor** [comment]
Merge please :)

File: lib/clipboard_image_paste/attachment_patch.rb (hunk 1 above)
Comment (Utopism) [reply lib/clipboard_image_paste/attachment_patch.rb]: You're right unloadable breaks auto-reloading.

File: README.textile [hunk 2]
```diff
@@ -152,3 +152,5 @@ Legend:
 [-] Bug fixed (we hope)
 [!] Important info
 </pre>
+
+!{width: 10%; height: 10%}https://compteur-visites.ennder.fr/sites/31/token/githubcip/image!
```
Comment (kou) [inline README.textile]: Is this change needed for this pull request?

Comment (Utopism) [reply README.textile]: Not at all, it's something I use to see if my plugin has some success.

Comment (kou) [reply README.textile]: Thanks.
I'm not a maintainer of this repository but I think that not related changes should be done in separated pull request in general terms.

Comment (Utopism) [reply README.textile]: Hello kou.
You're right, I'm going to remove the image link.

Comment (Utopism) [reply README.textile]: Done.

Comment (kou) [reply README.textile]: Thanks!

**Comment by AdamN1** [comment]
Hi,
can we merge this?

**Comment by petrlosa** [comment]
please merge this, this work on 4.0.3 OK

**Comment by jerome-salanhac** [comment]
work fine, thank a lot ! 👍

**Comment by Utopism** [comment]
> work fine, thank a lot ! +1 (redmine 4.0.3)

You're welcome.

**Comment by shanedobson-ibboost** [comment]
This is failing for me in Redmine 4.0.4. The install and migrate works perfectly well, but when I actually attempt to click on the button to use the plugin, nothing happens. Nothing in the log either.

**Comment by shanedobson-ibboost** [comment]
This is failing for me in Redmine 4.0.4. The install and migrate works perfectly well, but when I actually attempt to click on the button to use the plugin, nothing happens. Nothing in the log either. As well as that the button isn't rendering correctly - it appears below well outside the area where it normally does.

redmine

**Comment by shanedobson-ibboost** [comment]
Actually to add to that - the 'Add picture from clipboard' button appears at the bottom of all pages on Redmine after installing this

**Comment by scottfu001** [comment]
thanks a lot, worked on 4.0.4 stable.

**Comment by shanedobson-ibboost** [comment]
As above, it's not working from me on 4.0.4 stable

**Comment by Nico207** [comment]
This patch work in this environment:

> Redmine version : 4.0.5.stable.19319
>   Ruby version : 2.5.1-p57 (2018-03-29) [x86_64-linux-gnu]
>   Rails version : 5.2.3

Thanks a lot !

**Comment by Utopism** [comment]
> This is failing for me in Redmine 4.0.4. The install and migrate works perfectly well, but when I actually attempt to click on the button to use the plugin, nothing happens. Nothing in the log either. As well as that the button isn't rendering correctly - it appears below well outside the area where it normally does.
> 
> redmine

Hi, did tried to remove old assets with :
RAILS_ENV=production rails assets:clean
Do you have any other plugin that could interfere with this one ?

**Comment by dsardelic** [comment]
Seems to be working OK in the following environment: 
Screenshot from 2020-05-16 22-44-55.

**Comment by bviktor** [comment]
Redmine 4.1 has this feature built-in, so having this plugin is pretty pointless ever since.

**Comment by dsardelic** [comment]
> Redmine 4.1 has this feature built-in, so having this plugin is pretty pointless ever since.

Right you are, dear sir. Feature #3816, released in v4.1.0. Totally missed it.

**Comment by borisbrodski** [comment]
> Redmine 4.1 has this feature built-in, so having this plugin is pretty pointless ever since.

This plugins is much better:

* You can crop the images
* You can give a meaningfull names to the images

So please, keep it going.

**Comment by hiroshi24da** [comment]
I agree with borisbrodski.
In addition, feature #3816 works when text formatting is markdown,
but does't work with CKEditor.

**Comment by eric-it6** [comment]
Thanks man, I think this is the only version that still works under 4.0.x. Tested under 4.0.6.

**Comment by olivierb2** [comment]
+1, working for me on redmine 4.0.7 (Debian Buster Backports official packages)
